File: Step-Up/Recursion/9) SubsequencesString.py
def subSequences(idx,str,newstr):
    if idx >= len(str):
        res.append(newstr)
        return

    newstr += str[idx]
    subSequences(idx+1,str,newstr)

    newstr = newstr[:-1]
    subSequences(idx+1,str,newstr)

    return res


str = "abc"
res = []
subSequences(0,str,"")
print(res)

# Output : 
# ['abc', 'ab', 'ac', 'a', 'bc', 'b', 'c', '']

File: DSA implemented in Java/Anchal/Recursion/9) SubsequencesString.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    // Function to generate subsequences
    private List<String> subSequences(int idx, String str, String newstr) {
        List<String> res = new ArrayList<>();
        if (idx >= str.length()) {
            res.add(newstr);
            return res;
        }

        newstr += str.charAt(idx);
        res.addAll(subSequences(idx + 1, str, newstr));

        // Exclude the current character from the subsequence
        newstr = newstr.substring(0, newstr.length() - 1);

        res.addAll(subSequences(idx + 1, str, newstr));

        return res;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String str = "abc";
        List<String> res = solution.subSequences(0, str, "");
        System.out.println(res);
    }
}
